src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -225,13 +249,22 @@ private Boolean deploymentStatusChanged(Map<String, Object> deploymentDetails) {
         reported.put(STATUS_KEY, deploymentDetails.get(DEPLOYMENT_STATUS_KEY_NAME));
         reported.put(STATUS_DETAILS_KEY, statusDetails);
         reported.put(GGC_VERSION_KEY, deviceConfiguration.getNucleusVersion());
-
         return reported;
     }
 
     protected void shadowUpdated(Map<String, Object> desired, Map<String, Object> reported, Integer version) {
+        if (lastVersion != null && lastVersion > version) {
+            logger.atInfo().kv("SHADOW_VERSION", version)
+                    .log("Received an older version of shadow. Ignoring...");
+            return;
+        }
+        lastVersion = version;
+        //the reported section of the shadow was updated
+        if (reported != null && !reported.isEmpty()) {
+            syncShadowDeploymentStatus(reported);
```

Comment:
hmmm...don't you want to terminate the code execution here once `syncShadowDeploymentStatus` returns? Otherwise wouldn't it kick off another deployment?

Reply:
The device de-dupes deployment based on fleet config arn.

Reply:
Why not just exit here if that's right behavior? That way the code is also more readable. I don't see any benefit of relying the deployment to dedupe it unless there is a scenario when it's a new deployment while the `reported` is also non-null.

Reply:
A device can get notified of a deployment via two mechanisms:
1. When the device is online, it subscribes to /update/accepted topic to get notified of any changes to the shadow document.
2. When the device is offline for a while. When the device reconnects, it fetches the whole shadow document to see if it missed any deployment notification when it was offline. In this scenario the reported section and desired section can be non-null.

Because shadowUpdated() handles both the scenarios, we cant return after syncShadowDeploymentStatus.